eng/scripts/Invoke-GeneratAndBuildV2.ps1

```diff
@@ -8,33 +8,78 @@ param (
 
 $inputJson = Get-Content $inputJsonFile | Out-String | ConvertFrom-Json
 $swaggerDir = $inputJson.specFolder
+$swaggerDir = Resolve-Path $swaggerDir
 $swaggerDir = $swaggerDir -replace "\\", "/"
+[string[]] $inputFilePaths = $inputJson.changedFiles;
 $readmeFiles = $inputJson.relatedReadmeMdFiles
 $commitid = $inputJson.headSha
 $repoHttpsUrl = $inputJson.repoHttpsUrl
+$downloadUrlPrefix = $inputJson.installInstructionInput.downloadUrlPrefix
+[string] $autorestConfig = $inputJson.autorestConfig
 
-$generatedSDKPackages = New-Object 'Collections.Generic.List[System.Object]'
+$autorestConfigYaml = ""
+if ( $null -ne $autorestConfig -and $autorestConfig -ne "") {
+    $autorestConfig | Set-Content "config.md"
+    $autorestConfigYaml = Get-Content -Path .\config.md
+    $range = ($autorestConfigYaml | Select-String -Pattern '```').LineNumber
+    if ( $range.count -gt 1) {
+        $startNum = $range[0];
+        $lines = $range[1] - $range[0] - 1
+        $autorestConfigYaml = ($autorestConfigYaml | Select -Skip $startNum | Select -First $lines) |Out-String
+    }
+    Install-Module -Name powershell-yaml -Force -Verbose -Scope CurrentUser
+    Import-Module powershell-yaml
+    $yml = ConvertFrom-YAML $autorestConfigYaml
+    $requires = $yml["require"]
+    if ($requires.Count -gt 0) {
+        $readmeFiles = $requires
+    }
+}
 
-foreach ( $relateReadmeFile in $readmeFiles ) {
-    $readmeFile = $relateReadmeFile -replace "\\", "/"
-    
-    $sdkPath =  (Join-Path $PSScriptRoot .. ..)
-    $sdkPath = Resolve-Path $sdkPath
-    $sdkPath = $sdkPath -replace "\\", "/"
+$generatedSDKPackages = New-Object 'Collections.Generic.List[System.Object]'
```

Comment:
Tip: this is unnecessarily complicated. Just `$generatedSDKPackages = @()` accomplishes the same thing.

Reply:
The $generatedSDKPackages will be passed to the inner function as reference value, and in the inner function, it will add new item into it. if define it as `$generatedSDKPackages = @()`, it is an object, but it failed to add the item into it in inner function. So I need to defined it as an Collections.Generic.List

Reply:
You just use `$generatedSDKPackages += "new item"`, for example. PowerShell supports that for all collections by default. No need to call `Add` explicitly.

Reply:
Thanks @heaths I will try.

Reply:
Hello @heaths I have tried again. the normal list cannot be passed as reference, the item added to it in the function, and after back to caller, the list does not contain item.